cmake/host-tools.cmake

```diff
@@ -43,6 +43,15 @@ find_program(
   openocd
   )
 
+# jlink is an optional dependency
+
+find_program(
+  JLINK
+  jlink
```

Comment:
It is JLinkExe on Windows isn't it?

Perhaps more or different executable names should be given to 'find_program'.

Reply:
JlinkExe is on linux systems, jlink.exe is on windows,.. the name of the executable was solved by 
[12380](https://github.com/zephyrproject-rtos/zephyr/pull/12380)

I see my jlink changes are also in, purpose was to make other pull request

Reply:
Okay, I'm not sure what's going on. But we want to use 'find_program' instead of 

`DEFAULT_JLINK_EXE = 'JLink.exe' if sys.platform == 'win32' else 'JLinkExe'`

to find the appropriate JLINK_EXE.